resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt

```diff
@@ -113,7 +110,30 @@ class TemporaryResourceStorageService(
         return Path.of(tempDir.pathString, safeFileName)
     }
 
+    fun getMetadataValue(resourceStorageFieldId: String, metadataKey: String): String {
+        return getEnumFromKey(metadataKey).fold(
+            onSuccess = { enumKey ->
+                try {
+                    repository.getReferenceById(
+                        ResourceStorageMetadataId(
+                            fileId = resourceStorageFieldId,
+                            metadataKey = enumKey
+                        )
+                    ).metadataValue
+                } catch (e: EntityNotFoundException) {
+                    logger.warn("Resource $resourceStorageFieldId does not exist: ${e.message}")
+                    throw e
```

Comment:
This way, the error becomes part of the stacktrace:

```suggestion
                    throw IllegalStateException("Resource $resourceStorageFieldId does not exist", e)
```